Nahiyan_Dark mode on all modal in User Management

Nahiyan-16 opened the pull request:
# Description
All modals are now fully integrated with dark mode on the user management page

## Related PRS (if any):
This frontend PR is related to the dev backend

## Main changes explained:
- All modals have dark mode

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as owner user
5. go to dashboard→ Other Links -> User Management
6. verify all modals are working properly. 

## Screenshots or videos of changes:
![Screenshot 2024-05-08 130011](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/71025942/e6586f50-6539-4bbe-94c6-ca967ce0b102)
![Screenshot 2024-05-08 130034](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/71025942/498b1a22-9716-4a90-a6a4-e32954e3c99b)

Nahiyan-16 committed:
Dark mode on all modal in User Management

Ramaruva reviewed (changes requested):
Hi @Nahiyan-16  I have checked all modal in user management page all modals are in dark mode except for the change password modal I am not sure whether the change password have darkmode or not?
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/64008831/af999cc6-a9a6-4652-aa3d-336de85dae75)


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/64008831/eaca57f3-f149-4d96-88e5-0031fa0874fa

(https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/64008831/b92c80de-c84e-47ff-8fdd-d8496c6b2ce2)

and also I have one suggestion that can we make this icon "X" has white in dark mode instead of black because as a user I feel I am unable to see properly  "X" icon in dark mode

MinSun0814 reviewed (dismissed):
The updated dark mode looks fantastic and aligns with the aesthetic of the application. Nice work!
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/145076494/b7b873dd-c8a8-4fd3-a9b5-b24e276a11e7)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/145076494/7f33d52e-4ab8-48b5-9ed2-b6d72dc89bfa)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/145076494/06594fec-d017-4c2c-b365-8ba38d267666)

linkhanthtel reviewed (dismissed):
I reviewed your PR and all modals are working properly. Nice work! 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95618937/e314b610-75b3-4ae1-93e7-6bcb2df94751

Nahiyan-16 committed:
New Custom Header added

Nahiyan-16 commented:
> Hi @Nahiyan-16 I have checked all modal in user management page all modals are in dark mode except for the change password modal I am not sure whether the change password have darkmode or not? ![image](https://private-user-images.githubusercontent.com/64008831/329082258-af999cc6-a9a6-4652-aa3d-336de85dae75.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNzQ3NDgsIm5iZiI6MTcxNTI3NDQ0OCwicGF0aCI6Ii82NDAwODgzMS8zMjkwODIyNTgtYWY5OTljYzYtYTlhNi00NjUyLWFhM2QtMzM2ZGU4NWRhZTc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA5VDE3MDcyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhZDQyZWI5NTU2MGE4Y2E4N2ZiZmRmNTZjZWVjNTc4NzdmODMwMzZlOGVmMDYxNDc1MzY2YTM2Mzg5Zjc0MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.G7rFdqjtg6KBq-Mzvc1rZlutLJcRMDKvxEIYLkorce0)
> 
>  usermodel.mp4 
> (https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/64008831/b92c80de-c84e-47ff-8fdd-d8496c6b2ce2)
> 
> and also I have one suggestion that can we make this icon "X" has white in dark mode instead of black because as a user I feel I am unable to see properly "X" icon in dark mode

Fixed the issue regarding dark mode on password modal and added new custom header with the X mark being able to toggle from light to dark

Nahiyan-16 committed:
Update CustomHeader.jsx

Nahiyan-16 committed:
New file name for custom modal header

Nahiyan-16 committed:
Update EditableInfoModal.jsx

Ramaruva commented:
> > Hi @Nahiyan-16 I have checked all modal in user management page all modals are in dark mode except for the change password modal I am not sure whether the change password have darkmode or not? ![image](https://private-user-images.githubusercontent.com/64008831/329082258-af999cc6-a9a6-4652-aa3d-336de85dae75.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNzQ3NDgsIm5iZiI6MTcxNTI3NDQ0OCwicGF0aCI6Ii82NDAwODgzMS8zMjkwODIyNTgtYWY5OTljYzYtYTlhNi00NjUyLWFhM2QtMzM2ZGU4NWRhZTc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA5VDE3MDcyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhZDQyZWI5NTU2MGE4Y2E4N2ZiZmRmNTZjZWVjNTc4NzdmODMwMzZlOGVmMDYxNDc1MzY2YTM2Mzg5Zjc0MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.G7rFdqjtg6KBq-Mzvc1rZlutLJcRMDKvxEIYLkorce0)
> > usermodel.mp4
> > (https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/64008831/b92c80de-c84e-47ff-8fdd-d8496c6b2ce2)
> > and also I have one suggestion that can we make this icon "X" has white in dark mode instead of black because as a user I feel I am unable to see properly "X" icon in dark mode
> 
> Fixed the issue regarding dark mode on password modal and added new custom header with the X mark being able to toggle from light to dark

Hi I verified the latest changes now I am able to see dark mode for reset password modal great work on the X  mark also
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/64008831/b0d05074-cf0d-4dbc-a5e7-1474c2633269)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/64008831/4a2d2ba8-3b65-44aa-bbe9-c77d20b6acd3)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/64008831/63fb05cc-7de9-47e0-9d00-72a44ff255c5)

Ramaruva reviewed (dismissed):
latest changes are working

ZhangPeizhou commented:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95693786/6575370a-00c2-4c08-9a93-783035d1f292

Everything works as expected, nice work!

ZhangPeizhou commented:
I want to mention that when I click to view a user profile, it automatically switches to light mode, and I have to switch it back manually. I am not sure if it needs to be fixed here or not, just let you know.

nathcarnevalli reviewed (dismissed):
The Dark mode worked as expected in User Management

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69649453/01cebc5a-6119-4b66-8781-2fb0dc7f555a

hbodgal reviewed (dismissed):
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/145157488/dd4e534d-8e9d-44ea-b482-ade9c846ce43)

it works well.

abdel-lall commented on src/components/UserManagement/logTimeOffPopUp.jsx:
```diff
@@ -4,7 +4,6 @@ import moment from 'moment-timezone';
 import {
   Button,
   Modal,
-  ModalHeader,
```
This error was caused by the deletion of this import.
![Screenshot 2024-05-11 230729](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49083865/404311cd-b438-41e7-9622-e330181a2707)

abdel-lall reviewed (changes requested):
The Edit time-off request modal do not change color in dark mode.

![Screenshot 2024-05-11 230639](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49083865/c3cc69ed-c0b3-43c7-8747-70df652f1866)

Nahiyan-16 committed:
Dark mode on Edit time-off request modal and reverted custom modal header

Nahiyan-16 committed:
Merge branch 'development' into Nahiyan_Dark-Mode-on-User-Management-Page

Nahiyan-16 committed:
Added dark mode for new pop up

cgomezhub reviewed (approved):
well done!
All the popups works in dark mode.

https://www.loom.com/share/e8e1cd66176143be8c7f11b8f7a63e19?sid=467ac1c1-95ef-4bdb-afb6-f5f7c9a34f57

sucheta90 reviewed (commented):
Hello @Nahiyan-16,

After reviewing this PR, I see the feature works as expected. Please refer to the video. However, I noticed that refreshing the page resets the dark mode back to light mode. Is this the intended behavior?

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/101291283/75114daa-5e92-4c02-9300-98ab39f53659

Nahiyan-16 commented:
> Hello @Nahiyan-16,
> 
> After reviewing this PR, I see the feature works as expected. Please refer to the video. However, I noticed that refreshing the page resets the dark mode back to light mode. Is this the intended behavior?
> 
>  PR2238_review.mov

No this is not the intended behavior and will be dealt with in another PR.

peter6866 reviewed (approved):
All modals in user management work great in dark modal. 
<img width="716" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/58479595/6eebc98f-0d57-4ba7-8d89-bbce205122bc">
<img width="719" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/58479595/d3e4f096-0a28-48ed-88ca-76f977b04949">
<img width="445" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/58479595/129b7b2e-d318-434f-a471-82e846a29de5">
<img width="448" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/58479595/0f0f188e-984f-4264-8927-cb27935a2905">

one-community commented:
Thank you all, merging!
